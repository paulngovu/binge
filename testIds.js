export const TESTID_LOGIN_TITLE = 'login-title';
export const TESTID_LOGIN_USERNAME = 'login-username';
export const TESTID_LOGIN_PASSWORD = 'login-password';
export const TESTID_LOGIN_BUTTON = 'login-btn';
export const TESTID_LOGIN_ERROR_MSG = 'login-error-msg';
export const TESTID_LOGIN_ERROR_BUTTON = 'login-error-msg-btn';
