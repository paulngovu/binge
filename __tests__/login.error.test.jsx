/*
 * Testing for login error page
 */

import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import LoginError from '../pages/login/error';
import { TESTID_LOGIN_ERROR_BUTTON } from '../testIds';

describe('Login error page', () => {
  let tree;
  let getByTestId;

  beforeEach(() => {
    tree = renderer.create(<LoginError />).toJSON();
    ({ getByTestId } = render(<LoginError />));
  });

  test('Renders entire page', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Renders button to go back to login', () => {
    expect(getByTestId(TESTID_LOGIN_ERROR_BUTTON)).toBeTruthy();
  });
});
