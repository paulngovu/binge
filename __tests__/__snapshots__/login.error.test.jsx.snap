// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login error page Renders entire page 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 hjInHh"
>
  <header
    className="StyledBox-sc-13pk1d4-0 gpGtYH"
    display="flex"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 elclaf"
    >
      Binge
    </h1>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 itCgil"
    />
    <div
      alignself="end"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 wrGSV"
      />
    </div>
  </header>
  <div
    className="StyledBox-sc-13pk1d4-0 hHCfST"
  >
    <div
      className="jsx-3811088590 container"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 bYujHw"
      >
        <img
          className="StyledImage-sc-ey4zx9-0 fOsWNE"
          onLoad={[Function]}
          src="https://media.istockphoto.com/vectors/cartoon-cute-burger-icon-isolated-on-white-background-vector-id1142855781?k=20&m=1142855781&s=170667a&w=0&h=rBuRzFRXoS9jcc1FRL-rSexSP97wNsxhzJ-feGTeuJ4="
        />
      </div>
      <span
        className="StyledText-sc-1sadyjn-0 flpFwE"
      >
        You must log in.
      </span>
      <div
        className="StyledBox-sc-13pk1d4-0 gXhMXE"
      >
        <a
          className="StyledButton-sc-323bzc-0 emnQnE"
          data-testid="login-error-msg-btn"
          href="/login"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          Log In
        </a>
      </div>
    </div>
  </div>
</div>
`;
